deploy.py

```diff
@@ -2,58 +2,71 @@
 import json
 import os
 
-import yaml
-
-from utils import *
+from modules.splunkcloud import SplunkCloudConnector
+from modules.aws_s3 import AwsS3Connector
+from modules.apps_processing import AppFilesProcessor
 
 # FOR LOCAL TESTING
 # from dotenv import load_dotenv
 # load_dotenv(dotenv_path="local.env")
 
+SPLUNK_USERNAME = os.getenv("SPLUNK_USERNAME")
+SPLUNK_PASSWORD = os.getenv("SPLUNK_PASSWORD")
+SPLUNK_TOKEN = os.getenv("SPLUNK_TOKEN")
+
+AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID")
+AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY")
+
+
 def main():
     if len(sys.argv) != 2:
         print("Usage: python script.py <path_to_yaml_file>")
         sys.exit(1)
 
     yaml_file_path = "environments/" + sys.argv[1] + "/deployment.yml"
 
+    # Initiate deployment report
     deployment_report = {}
 
-    try:
-        data = read_yaml(yaml_file_path)
-    except FileNotFoundError:
-        print(f"Error: The file '{yaml_file_path}' was not found.")
-    except yaml.YAMLError as e:
-        print(f"Error parsing YAML file: {e}")
-        sys.exit(1)
+    # Initiate AppFilesProcessor object
+    app_processor = AppFilesProcessor(yaml_file_path)
+
     ### 1. Validate data and retrieve all apps listed in deployment.yml from S3 ###
-    private_apps, splunkbase_apps = validate_data(data)
+    data, private_apps, splunkbase_apps = app_processor.validate_data()
     # List all apps in yaml file and then their S3 bucket
     if private_apps:
         apps = data.get("apps", {}).keys()
         s3_buckets = [data["apps"][app]["s3-bucket"] for app in apps]
         app_directories = [data["apps"][app]["source"] for app in apps]
     target_url = data["target"]["url"]
-    # Download all apps from S3
+
+    # Initiate AwsS3Connector object
+    s3_connector = AwsS3Connector(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)
```

Comment:
Since we only need to interact with S3 to download files, I'd avoid creating the wrapper of the wrapper (boto3). It only increases complexity. Besides, I don't think you need to pass AWS parameters, since already defined as environment variables (see [docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-environment-variables)) 

Something like the following should work, without forgetting to: 
* Add `import boto3` at the top of this file
* Remove lines 17 and 18 of this file

```suggestion
    s3_connector = boto3.client("s3")
```